File: app/src/main/java/com/song/deviceinfo/info/MapsInfo.java
package com.song.deviceinfo.info;

import android.content.Context;

import com.song.deviceinfo.utils.LogUtils;

import java.io.BufferedReader;
import java.io.FileReader;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by chensongsong on 2021/9/10.
 */
public class MapsInfo {

    public static List<String> getMapsInfo(Context context) {
        List<String> list = new ArrayList<>();
        String mapsFilename = "/proc/" + android.os.Process.myPid() + "/maps";
        BufferedReader bufferedReader = null;
        FileReader fileReader = null;
        try {
            fileReader = new FileReader(mapsFilename);
            bufferedReader = new BufferedReader(fileReader);
            String line;
            while ((line = bufferedReader.readLine()) != null) {
                LogUtils.d("maps line: " + line);
                list.add(line);
            }
        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            try {
                if (bufferedReader != null) {
                    bufferedReader.close();
                }
            } catch (Exception e) {
                e.printStackTrace();
            }
            try {
                if (fileReader != null) {
                    fileReader.close();
                }
            } catch (Exception e) {
                e.printStackTrace();
            }

        }
        return list;
    }

}

File: app/src/main/java/com/song/deviceinfo/info/MapsInfo.kt
package com.song.deviceinfo.info

import android.content.Context
import android.os.Process
import com.song.deviceinfo.utils.LogUtils.d
import java.io.BufferedReader
import java.io.FileReader

/**
 * Created by chensongsong on 2021/9/10.
 */
object MapsInfo {
    fun getMapsInfo(context: Context?): List<String> {
        val list: MutableList<String> = ArrayList()
        val mapsFilename = "/proc/" + Process.myPid() + "/maps"
        var bufferedReader: BufferedReader? = null
        var fileReader: FileReader? = null
        try {
            fileReader = FileReader(mapsFilename)
            bufferedReader = BufferedReader(fileReader)
            var line: String
            while ((bufferedReader.readLine().also { line = it }) != null) {
                d("maps line: $line")
                list.add(line)
            }
        } catch (e: Exception) {
            e.printStackTrace()
        } finally {
            try {
                bufferedReader?.close()
            } catch (e: Exception) {
                e.printStackTrace()
            }
            try {
                fileReader?.close()
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
        return list
    }
}
